language/move-lang/src/expansion/dependency_ordering.rs

```diff
@@ -416,6 +494,18 @@ fn spec_block_member(context: &mut Context, sp!(_, sbm_): &E::SpecBlockMember) {
         M::Let { def: e, .. } | M::Include { exp: e, .. } | M::Apply { exp: e, .. } => {
             exp(context, e)
         }
-        M::Variable { .. } | M::Pragma { .. } => {}
+        M::Pragma { properties } => {
+            for prop in properties {
+                let pragma = &prop.value;
+                if pragma.name.value == "friend" {
```

Comment:
This is actually a very tricky one.

The `pragma friend = <address::module_name::function_name>` notion exists before we implement the `friend` scheme. And it may still have a use case, that is, to friend the `Genesis` module.

Right now, the `Genesis` module is not published on chain. Therefore, we cannot really declare `friend Genesis;` in the module, because that will lead to linking error. But the prover side still needs to know that `Genesis` is a friend (to verify global invariants). So, the `pragma friend = ...` syntax might need to stay. And for that, we need to add the module that is declared as a friend in the `dependency_summary`.

Reply:
Oof, alright have at it I suppose.

But I would definitely add what you just said as a comment

Reply:
Will do!